buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -617,6 +629,12 @@ class BuildPlugin implements Plugin<Project> {
         }
     }
 
+    private static void assertRepositoryURIUsesHttps(final Project project, final URI uri) {
+        if (uri != null && uri.toURL().getProtocol().equals("http")) {
+            throw new GradleException("repository on project [" + project.name + "] is using http for artifacts on [" + uri.toURL() + "]")
```

Comment:
We should use `project.path` instead of `project.name` even though the full path will likely appear elsewhere in the exception message and/or stacktrace. We have a lot of ambiguous subproject names.